tour_model_eval/viz_bin_cluster.py

```diff
@@ -0,0 +1,79 @@
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy
+import emission.core.get_database as edb
+import logging
+import math
+import emission.storage.timeseries.abstract_timeseries as esta
+import emission.analysis.modelling.tour_model.cluster_pipeline as eamtc
+import emission.analysis.modelling.tour_model.similarity as similarity
+import emission.analysis.modelling.tour_model.cluster_pipeline as pipeline
+import emission.analysis.modelling.tour_model.featurization as featurization
+
+# imports for visualization code
+import folium
+import branca.colormap as cm
+
+
+def bins_map(bins, sel_bin_indices, trips):
+    # Plot all the bin trips on the map, use different colors for different bins.
+    # Each color represents one bin based on index
+    # Choose the first start point from the trips to locate the map for convenience
+
+    # The function takes three parameters:
+    # - bins: bins list above cutoff after running pipeline.remove_noise
+    # - sel_bin_indices: the list of indices of selected bins (set to None for plotting all bins)
+    # - trips: the trips after running pipeline.read_data
+
+    # In order to plot trips correctly, map index should be len(bins)+1,
+    # so that the color steps length is equal to the bins number. For bin index which is 0,
+    # will be plotted in color which is in index 1.
+    color_map = cm.linear.Set1_07.to_step(len(bins), index=[i for i in range(len(bins)+1)])
+    map = folium.Map(location=[trips[0].data.start_loc["coordinates"][1], trips[0].data.start_loc["coordinates"][0]],
+                   zoom_start=12, max_zoom=30, control_scale=True)
```

Comment:
and this

Reply:
Do you mean the `zoom_start=12, max_zoom=30`? 
I don't think I can.
```zoom_start (int, default 10) – Initial zoom level for the map.```--from folium documentation
I have tried many times to set it to 12, in order to fit most of the graphs (on all data and above cutoff)

Reply:
no, I mean that you should call `fit_bounds` to set the zoom dynamically.